eap-latest-setup.md

```diff
@@ -1697,9 +1697,79 @@ directory with:
 
     mkdir -p /mnt/registry
 
+### Creating user account for registry
+Each pod runs under particular service account. Three such accounts are created for you
+by default in each project:
+
+    os get serviceaccounts
+    NAME       SECRETS
+    builder    2
+    default    2
+    deployer   2
+
+`builder` service account is required by builder pods and `deployer` by
+deployment pods. The `default` is used for all the other pods unless
+overridden.
+
+Each service account can be given security context constraints (SCC). They
+allow to specify SELinux context for pods, limit users used to run them etc.
+To list them, type:
+
+    oc get scc
+    NAME         PRIV      CAPS      HOSTDIR   SELINUX     RUNASUSER
+    privileged   true      []        true      RunAsAny    RunAsAny
+    restricted   false     []        false     MustRunAs   MustRunAsRange
+
+By default, newly created pods are *restricted* which protects users from compromising
+the system. However, restricted pod does not allow you to mount host directory. The
+registry pod needs be run under *privileged* service account. Since all currently
+existing have already their use, let's set up a new one as `root`:
+
+    oc create -f - <<EOF
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: registry
+    EOF
+
+Next step is to make it *privileged*. This is achieved by adding service
+account's user name to a list of users of *privileged* SCC. Such a name looks
+like this:
+
+    system:serviceaccount:<namespace>:<name>
+
+Where `<namespace>` is `default` because we set up registry in the *default*
+namespace and `<name>` is whatever we used in previous `oc create` command.
+
+Type following command and add the user name:
+
+    oc edit scc priveliged
```

Comment:
Host Dir is incredibly dangerous - its effectively root in some cases.
It's less dangerous than priv, but only slightly.

On Jul 17, 2015, at 9:57 AM, Jordan Liggitt <notifications@github.com>
wrote:

In eap-latest-setup.md
<https://github.com/projectatomic/atomic-enterprise-training/pull/21#discussion_r34891001>
:

> +    metadata:
> +      name: registry
> +    EOF
> +
> +Next step is to make it *privileged*. This is achieved by adding service
> +account's user name to a list of users of *privileged* SCC. Such a name looks
> +like this:
> +
> +    system:serviceaccount:<namespace>:<name>
> +
> +Where `<namespace>` is `default` because we set up registry in the *default*
> +namespace and `<name>` is whatever we used in previous `oc create` command.
> +
> +Type following command and add the user name:
> +
> +    oc edit scc priveliged

two questions:
1. Is host dir what we want, or do we want persistent volumes?
2. @pweil- <https://github.com/pweil->, is privileged necessary just to get
host dir? I'd rather have an scc that allows host dir specifically and add
the registry service account to that

—
Reply to this email directly or view it on GitHub
<https://github.com/projectatomic/atomic-enterprise-training/pull/21/files#r34891001>
.